### ISSUE TITLE
Dark theme - Disabled stages not colored in Pipeline State window

### ISSUE THREAD

**Issue by akbyrd** [baldurk/renderdoc#1141]
## Description

It looks like the Dark theme doesn't have separate colors for enabled and disabled stages in the Pipeline State window.

![renderdoc - light](https://user-images.githubusercontent.com/484784/47461047-b533c280-d794-11e8-9c7a-c2cc64073fc1.png)

![renderdoc - dark](https://user-images.githubusercontent.com/484784/47461050-b6fd8600-d794-11e8-8811-670b68bc0b45.png)

## Environment

* Version v1.1 (built from c7d5aa86) (I wish this text were copy-able)
* Windows 10 Enterprise Version 1709 Build 16299.666
* D3D12

**Comment by baldurk** [comment]
I'm not sure there's a _ton_ I can actually do about this. The pipeline flowchart just uses standard colours from the current theme's palette, so there is actually a difference it's just so small that it's basically imperceptible without using a colour dropper on the screenshot. The same kind of thing is visible on the active vs inactive tabs in the docked panels.

What you're seeing falls out of the dark palette having a lot lower contrast for many things by virtue of being a dark palette. That's not something specific to RenderDoc's dark style, I see the same thing if I use an operating system dark style like KDE's.

The best I can do I think is add an extra indicator to disabled stages by picking a different outline colour, since that does have more contrast on the dark theme since it's normally white and is usually dulled to grey for inactive things. How does this look?

![image](https://user-images.githubusercontent.com/661798/47490744-4d878100-d841-11e8-9b46-ed666bc0e001.png)

![image](https://user-images.githubusercontent.com/661798/47490656-187b2e80-d841-11e8-8a4d-fda8f1728331.png)

It looks a bit odd now on the normal theme, but since it's a custom control and not something more natural like a checkbox or button which you'd expect to have a consistent border, I think you get away with it.

(Also side note - you don't need to list out the full commit hash if you're running a stable version, that's only needed if you're running a nightly build and the date would also be fine)

**Comment by akbyrd** [comment]
Is it possible to use the darker outline color as the fill instead? Like you said, the outline makes the light version look worse (almost feels blurry). Or maybe using opacity to fade disabled stages slightly?

**Comment by baldurk** [comment]
No, the new outline colour there is `WindowText` with the `Disabled` role, intended to be used for text, outlines, etc. It's not a fill colour.

Like I say the control itself doesn't have custom colouring code based on the theme because the theme is arbitrary if you select 'native'. I can't pick specific RGB colours so I'm limited to what colours are available in the defined Qt palette, and I can't make any assumptions about what the different colours are in relation to each other. Otherwise I run the risk of creating a case where on some themes there isn't enough contrast between the actual stage text and the background.

In the end I don't think there are any great compromises, as you're pretty much asking for adding contrasting colours when using a theme that by definition has had a lot of contrast removed.

**Comment by baldurk** [comment]
I experimented with this more, and with some re-jigging I found an option that might be a better trade-off. I changed it so the fill colour was `Disabled / AlternateBase` which is still dark but not quite as dark (and has the advantage of improving the text contrast on the normal theme), and still allows a disabled border colour while keeping a darker border than fill colour.

On the dark theme you still don't much of an indication because the background colour can't change, but you do get the disabled border as an indicator. As I mentioned the dark theme has deliberately eliminated most options for contrasting colours as it's mostly just various greys on black, but it's perhaps better. This should also be safe with any system theme as well:

![image](https://user-images.githubusercontent.com/661798/47651106-2c44ce80-db7a-11e8-9488-abe46989e1f2.png)

![image](https://user-images.githubusercontent.com/661798/47651122-3797fa00-db7a-11e8-80dc-8c803d35a23c.png)
